### DISCUSSION TITLE
Project 1 Feedback - Team Epsilon

### DISCUSSION THREAD

**Discussion by pbrm4** [airavata-courses/neo#69] [General]
Hi,
The installation was very straightforward and fast. Cannot say the same about loading the plots. 
I was able to add multiple widgets to show me different plots or places but as soon as I requested another the previous request got canceled.
Please refer to the image.

![image](https://user-images.githubusercontent.com/12838992/153333681-d1f7d398-899b-49a4-a738-3b89ab722962.png)

--Pramod
Team Epsilon

**Comment by chanakyaken** [comment]
Hi Team,

Just to add what Pramod said above, It was nice to see a loader until the data is fetched but it took me near about 2-3 mins to load the plots.
I also liked the History page especially the history count and page wise recents segregation.

--Chanakya
Team Epsilon

**Comment by niravraje** [comment]
Thanks Team Epsilon! 
We'll look into the multiple requests point as this is new to us.
Regarding the loading time, it seems to be occurring on Mac (M1) (especially with 8 GB memory) where the default setting for Docker resource limits is 2 GB and all microservices are only being limited to utilize 2 GB of RAM.
On a Windows machine with 16 GB RAM and a setting of 12 GB permitted for Docker, the visualization generation takes less than 20 seconds.

**Comment by rajdeepc2792** [comment]
Thanks Team Epsilon for finding the multiple request issue,
It is because of classic Angular's NgRx effects handling problem. 

Last request Observable is getting cancelled as we used RxJS "Switch Map" in place of "Merge Map".
Performing above change will allow us to place multiple request.
